Firestore/core/test/unit/util/executor_test.cc

```diff
@@ -379,6 +379,55 @@ TEST_P(ExecutorTest, DestructorAvoidsDeadlockWhenDeletingSelf) {
   ASSERT_EQ(result, "12");
 }
 
+TEST_P(ExecutorTest, DipsoseBlocksTaskSubmission) {
+  // Verify there's no crash for an idempotent invocation.
+  executor->Dispose();
+  executor->Dispose();
+
+  Expectation ran;
+  executor->Execute(ran.AsCallback());
+
+  auto status = ran.get_future().wait_for(Executor::Milliseconds(50));
+  ASSERT_EQ(status, std::future_status::timeout);
+}
+
+TEST_P(ExecutorTest, DipsoseBlocksConcurrentTaskSubmission) {
+  Expectation allow_destruction;
+  Expectation blocking_task_running;
+
+  // Run a task that blocks. These cause Dispose to block.
+  executor->Execute([&] {
+    blocking_task_running.Fulfill();
+    Await(allow_destruction);
+  });
+
+  Await(blocking_task_running);
+
+  // Run `Dispose`. This will block because there's a task pending.
+  Expectation dispose_running;
+  Expectation dispose_complete;
+  Async([&] {
+    dispose_running.Fulfill();
+    executor->Dispose();
+    dispose_complete.Fulfill();
+  });
+
+  Await(dispose_running);
+  Expectation execute_running;
+  Expectation ran;
+  Async([&] {
+    execute_running.Fulfill();
+    executor->Execute(ran.AsCallback());
```

Comment:
Consider adding a brief comment here, similar to the ones above, to explain the intent of this `Execute`.

Reply:
This was actually pretty critical--thanks for pushing on this.